packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Providers/ModelProvider.cs

```diff
@@ -399,9 +399,23 @@ protected override PropertyProvider[] BuildProperties()
                             outputProperty.Modifiers |= MethodSignatureModifiers.Virtual;
                         }
                     }
-                    var baseProperty = baseProperties.GetValueOrDefault(property.Name);
-                    if (baseProperty is not null)
+
+                    if (baseSystemProperties.TryGetValue(property.Name, out var baseSystemProperty))
+                    {
+                        if (DomainEqual(baseSystemProperty, property, isBaseSystemProperty: true))
```

Comment:
BaseType from Azure.ResourceManager is not handling the property generation same as MTG, i.e. adding backing field and virtual for the derived property.
So, we need to handle it differently.

Once we generate Azure.ResourceManager with MTG, we can remove this special handling.